maptime/README.md

```diff
@@ -0,0 +1,53 @@
+# Maptime
+
+Maptime is a Golang linter that detects usages of maps where the key of the map contains an instance of time.Time.
+
+While an instance of time.Time can be safely stored as part of the key of a map, it's very easy to introduce subtle bugs this way and it's often much safer to use something like an int64 to store a unix timestamp at nanosecond resolution instead.
+
+For detailed information on why storing a time.Time as a map key can be
+dangerous, read [this section](https://golang.org/src/time/time.go?#L101) of
+the golang documentation for the time package.
+
+Note that this issue can be particularly troublesome for projects upgrading to Golang 1.9 because in previous versions of Golang, two instances of time.Time for the same moment of time would only not be == to each other if they represented different timezones, whereas in Golang 1.9 and later they can also not be == to each other if one contains a monotonic bit and the other does not.
+
+## Gometalinter integration
+
+maptime is designed to integrate with [gometalinter](https://github.com/alecthomas/gometalinter). To add it to the list of active linters, make sure maptime is installed, and then modify the `.metalinter.json` file to add "maptime" to the "Enable" array and add the following entry in the "Linters" object: `"maptime": "maptime:PATH:LINE:COL:MESSAGE"`
+
+Example:
+
+```json
+{
+  "Linters": {
+    "maptime": "maptime:PATH:LINE:COL:MESSAGE"
+  },
+  "Enable": ["maptime"],
+}
+```
+
+## Installation
+
+```bash
+go get https://github.com/m3db/build-tools/maptime
+```
+
+## Usage
+
+```golang
+maptime ./...
```

Comment:
could you add a flag that skips `vendor/` by default, and add support for build tags

Reply:
Could you expand on this? I looked through the source code for some other gometalinter plugs (including unused and gosimple which I see you passing a -tags variable to in our metalinter.json) but none of them seem to have any explicit code that handles vendor directors / build tags, whereas gometalinter will handle that for you

Reply:
Yeah most of the linters don't skip `vendor/` so you end up having to either do that manually, something like: `./linter $(go list ./... | grep -v /vendor/)`. Or rely on the metalinter. Figured if people are going to run this by hand it's nice to not require that. 

Re: build-tags. I don't know how the AST parser handles type compilation when build tags are present. For instance, `unused` doesn't work for `m3db/m3db/integration` unless you specify the build-tags. 

<details>
<summary>`unused` on m3db integration tests w/ and w/o build tags</summary>

~/code/gocode/src/github.com/m3db/m3db master ⇣ 3m 55s
❯ unused --tags integration ./integration


~/code/gocode/src/github.com/m3db/m3db master ⇣
❯ unused ./integration
integration/client.go:42:6: func tchannelClient is unused (U1000)
integration/client.go:54:6: func tchannelClientWriteBatch is unused (U1000)
integration/client.go:76:6: func tchannelClientFetch is unused (U1000)
integration/client.go:86:6: func tchannelClientTruncate is unused (U1000)
integration/client.go:95:6: func tchannelClientHealth is unused (U1000)
integration/client.go:100:6: func m3dbAdminClient is unused (U1000)
integration/client.go:105:6: func m3dbClientWriteBatch is unused (U1000)
...
</details>

Reply:
Ok I added the appropriate flags